botorch/acquisition/objective.py

```diff
@@ -576,7 +584,17 @@ def forward(self, samples: Tensor, X: Optional[Tensor] = None) -> Tensor:
             A `(sample_size * num_samples) x batch_shape x N`-dim Tensor of
             objective values sampled from utility posterior using `pref_model`.
         """
-        post = self.pref_model.posterior(samples)
+        if samples.dtype == torch.float32 and any(
+            [d == torch.float64 for d in self.pref_model.dtypes_of_buffers]
+        ):
+            warnings.warn(
+                _get_learned_objective_pref_model_mixed_dtype_warn(),
+                InputDataWarning,
+            )
+            samples = samples.to(torch.float64)
+
+        posterior_ = self.pref_model.posterior
+        post = posterior_(samples)
```

Comment:
Is there anything wrong with this? 
```suggestion
        post = self.pref_model.posterior(samples)
```